tests/test_trial.py

```diff
@@ -1271,6 +1271,19 @@ def test_create_trial(state: TrialState) -> None:
     assert trial.datetime_start is not None
     assert (trial.datetime_complete is not None) == (state is None or state.is_finished())
 
+    if state is None:
+        # Raise `ValueError` when both `value` and `values` are `None`.
+        with pytest.raises(ValueError):
+            create_trial(state=state, value=None, values=None)
+        # Raise `ValueError` when both `value` and `values` are passed.
+        with pytest.raises(ValueError):
+            create_trial(state=state, value=value, values=[value, value])
+        # Raise `ValueError` when either `params` or `distributions` is `None`.
+        with pytest.raises(ValueError):
+            create_trial(state=state, value=value, params=params, distributions=None)
+        with pytest.raises(ValueError):
+            create_trial(state=state, value=value, params=None, distributions=distributions)
```

Comment:
How about parametrizing it and defining a separate function. It's quite independent of the above logic (`trial`) and would make test failures easier to comprehend.

Reply:
I think it's a bit too early to separate the function as the API is still experimental, and the function should cover the whole bunch of  `create_trial` cases as it's named "create_trial". 

Also, separating functions is not always great as it increases the number of lines, and test_trial.py is already large enough to lose comprehensiveness.

Reply:
It's not a strong opinion so let's then leave it as is.